jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py

```diff
@@ -269,13 +273,44 @@ def _on_flux_unit_changed(self, msg):
         else:
             self.flux_or_sb_selected = 'Surface Brightness'
 
+        # Always send a surface brightness unit to contours
+        if self.flux_or_sb_selected == 'Flux':
+            yunit = self._append_angle_correctly(yunit, self.angle_unit.selected)
+        self._find_and_convert_contour_units(yunit=yunit)
```

Comment:
what if `_find_and_convert_contour_units` just observed `sb_unit_selected` which already handles this logic?

Reply:
we discussed that unit conversion probably needs some code moving soon anyways, so I'm fine deferring this until then